src/kOS.Safe/Encapsulation/Structure.cs

```diff
@@ -337,20 +337,10 @@ public static Structure FromPrimitiveWithAssert(object value)
 
         public static object ToPrimitive(object value)
         {
-            var scalarValue = value as ScalarValue;
-            if (scalarValue != null)
+            var primitive = value as PrimitiveStructure;
+            if (primitive != null)
             {
-                return scalarValue.Value;
-            }
-            var booleanValue = value as BooleanValue;
-            if (booleanValue != null)
-            {
-                return booleanValue.Value;
-            }
-            var stringValue = value as StringValue;
-            if (stringValue != null)
-            {
-                return stringValue.ToString();
+                return primitive.ToPrimitive();
```

Comment:
@hvacengi this is one of those hanging comments I was referring to in slack.  I don't know if you have work you wanted to do with this before I allow this to get merged.